examples/upgradeable/v1/src/test.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(test)]
+
+extern crate std;
+
+use contract_v2::Data;
+use soroban_sdk::{testutils::Address as _, Address, BytesN, Env};
+
+use crate::contract::{ExampleContract, ExampleContractClient};
+
+mod contract_v2 {
+    use crate::test::{MigrationData, RollbackData};
+
+    soroban_sdk::contractimport!(file = "../testdata/upgradeable_v2_example.wasm");
+}
+
+fn install_new_wasm(e: &Env) -> BytesN<32> {
+    e.deployer().upload_contract_wasm(contract_v2::WASM)
+}
+
+type MigrationData = Data;
+type RollbackData = ();
+
+#[test]
+fn test_upgrade() {
```

Comment:
I'll give another look to this test after reading the sourcecode (started reviewing first from test files). And currently, I don't have a good idea on what's happening in here. There may be other people who will take a look first at test files to quickly grasp what this code is about. Maybe we can put some inline comments here and there in this test, explaining what's happening and what's the expected behavior for the asserts.